### PR TITLE
samples: openthread: Fixed bugs in coap_client sample

### PR DESCRIPTION
**Pull request by MarekPorwisz**
Fixed MED/SED race condition when powering thread.
Fixed uart states being incorrect after power up.
Fixed radio driver on/off controler race condition.

Signed-off-by: Marek Porwisz <marek.porwisz@nordicsemi.no>

### CODE REVIEW

**Comment by MarekPorwisz** [comment]
Marked DNM as more bugfixes will come from upstream - manifest will be updated

File: samples/openthread/coap_client/src/coap_client_utils.c [hunk 1]
```diff
@@ -190,33 +188,21 @@ static void toggle_minimal_sleepy_end_device(struct k_work *item)
...
+		on_mtd_mode_toggle(mode.mRxOnWhenIdle);
 	}
+}
 
+static void update_device_state(void)
+{
+	struct otInstance *instance = openthread_get_default_instance();
+	otLinkModeConfig mode = otThreadGetLinkMode(instance);
 	on_mtd_mode_toggle(mode.mRxOnWhenIdle);
+
```
Comment (rlubos) [inline samples/openthread/coap_client/src/coap_client_utils.c:205]: Unneeded empty line.

File: samples/openthread/coap_client/src/coap_client.c [hunk 2]
```diff
@@ -84,9 +87,21 @@ static void on_ot_disconnect(struct k_work *item)
...
 
-static void on_mtd_mode_toggle(uint32_t val)
+static void on_mtd_mode_toggle(uint32_t med)
 {
-	dk_set_led(MTD_SED_LED, val);
+#if IS_ENABLED(CONFIG_DEVICE_POWER_MANAGEMENT)
+	struct device *cons = device_get_binding(CONSOLE_LABEL);
+
+	if (med) {
+
```
Comment (rlubos) [inline samples/openthread/coap_client/src/coap_client.c:96]: Unneeded empty line.

**Review by rlubos — COMMENTED** [review]
Looks good so far, just some nits.

**Comment by MarekPorwisz** [comment]
I removed DNM, seems that upstream changes are not cherry-pickable. It is ether-all or none as the hal_nordic comes via zephyrs yaml and it already has dependencies to other commits eg. struct device parameter is now const. Other fixes shall come with the upmerge of zephyr. (hopefully within couple of days)

**Review by pszkotak — APPROVED** [review]
Thanks Marek!

**Review by rlubos — APPROVED** [review]

**Pull request merged**